chall05/vsempere.c

```diff
@@ -0,0 +1,61 @@
+# include <stdlib.h>
+
+static size_t		ft_strlen(const char *s)
+{
+	size_t	i;
+
+	i = 0;
+	while (s[i] != 0)
+		i++;
+	return (i);
+}
+
+static char		*ft_strstr(const char *big, const char *little)
+{
+	size_t		i;
+	size_t		j;
+
+	i = 0;
+	j = 0;
+	while (big[i])
+	{
+		while (big[i + j] == little[j] && big[i + j])
+			j++;
+		if (little[j] == '\0')
+			return ((char *)(big + i));
+		j = 0;
+		i++;
+	}
+	return (NULL);
+}
+
+static char ft_check_exception(char *str)
+{
+	char	*exc;
+	char	*f;
+	size_t	strl;
+
+	exc = "sleigh stein fahrenheit deifies either nuclei reimburse ancient juicier societies";
+	if (!(f = ft_strstr(exc, str)))
+		return (0);
+	strl = ft_strlen(str);
+	return (((f > exc && *(f - 1) == ' ') || f == str) && (f[strl] == '\0' || f[strl] == ' ') ? 1 : 0);
+}
+
+int			ft_ie_except_after_c(char* str)
+{
+	char	*f;
+	
+	if (!str)
+		return (1);
```

Comment:
It is my opinion, but you don't have to protect from `NULL` when making such a function. A function is not like a program, a function CAN crash. Let me explain myself:  
Here this function has a defined behavior:
- It takes a word as input (lowercased although I think we forgot to explicitly specify it)
- It MUST return `1` when the rule is respected
- It MUST return `0` when the rule is not respected

But it is not specified what you must or should do when the parameter isn't allocated, or isn't a word... So then, if an unallocated string causes a segmentation fault, it is not your problem, the user should have passed an allocated string.  
Which brings me to this `NULL` correspond to `0x0`, you neither can write or read the data present there. So why would you protect this specific memory zone? And not another like `0x1` which can't be written nor read either. But this is yet a subjective point of view.  
To be more objective, `NULL` is not a word, it isn't even a string! So how could you decide if it should return `0` or `1`? Both return values would be correct, as it is an undefined behavior. And well, crashing is also correct, in an undefined behavior.

Beware, what I am saying only applies to separated functions like this, when you do a program, a website, a videogame, or anything of the sort, there should not be undefined behaviors, because then it would be a bug!